contribute/design/1.x/apis/cstor_volume.md

```diff
@@ -0,0 +1,320 @@
+# Move `CStorVolumes`, `CStorVolumeReplica` resources Schema to version `v1`
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [Non-Goals](#non-goals)
+* [Schema Proposal](#volume-schema-proposal)
+* [Risk and Mitigations](#risks-and-mitigations)
+* [CRD Migration and Graduation](#graduation-criteria)
+
+## Introduction
+
+This proposal highlights the limitations of current `CStorVolumes` and `CStorVolumeReplicas` schema and
+proposes improvements.
+
+# Goals
+
+The major goal of this document is to freeze the APIs schema for `CStorVolumes` and `CStorVolumeReplica`.
+Apart from this the document focuses on following aspects too:
+
+- Migrating the `CStorVolumes` and `CStorVolumeReplica` CRD to `v1` apiversion.
+- Identify the breaking and upgrade changes and documentation around that.
+- List of possible high level status fields on `CStorVolumes` and `CStorVolumeReplica`.
+
+# Non-Goals
+
+- Introduction of new fields in the `CStorVolumes` and `CStorVolumeReplica` to deliver feature etc.
+- List of possible events that should be recorded on `CStorVolumes` and `CStorVolumesReplicas`.
+- List of conditions that should be put on the `CStorVolumes` and `CStorVolumesReplicas` CRs.
+
+# Volume Schema Proposal
+
+## `CStorVolumes` APIs
+
+```go
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CStorVolume describes a cstor volume resource created as custom resource
+type CStorVolume struct {
```

Comment:
suggestion: It may be better to add the existing version of the cStorVolume and cStorVolumeReplica files - not embedded but as separate files and then compare the changes? It is a little hard now to know what exactly is changing from existing cStorVolume and cStorVolumeReplica which are already in beta.

Reply:
@kmova there is section called **change existing types** for the above info, however there is  not much changes in CStorVolume and CStorVolumeReplicas .
Let me know if i need to add the older APIs for the references.